src/error.rs

```diff
@@ -56,6 +81,10 @@ impl error::Error for Error {
             HeaderValueShared(ref e) => e.description(),
         }
     }
+
+    fn cause(&self) -> Option<&error::Error> {
+        Some(self.inner_ref())
```

Comment:
This should be `self.inner_ref().cause()` I believe.

Reply:
All of the current inner error types return `None` (e.g. don't implement `cause` themselves) so simply removing my impl., would have the same external effect. Or would you prefer to explicitly implement it this way, for future proofing sake?

Reply:
Implemented as the inner cause.